core/tauri/src/endpoints/shell.rs

```diff
@@ -31,13 +38,61 @@ fn command_child_store() -> &'static ChildStore {
   &STORE
 }
 
-#[derive(Debug, Clone, Deserialize)]
+#[derive(Debug, Clone, Serialize)]
+#[serde(tag = "event", content = "payload")]
+#[non_exhaustive]
+enum JSCommandEvent {
```

Comment:
I had to introduce this extra type to get serde serialize it into javascript event properly. Couldn't not use `CommandEvent` because it now contains `Vec<u8>` which would always serialize to `number[]`. 

Maybe you have some ideas how I can improve this?

Reply:
that's fine